### PR TITLE
Add a class for storing cluster-wide parameters

### PR DESCRIPTION
**Pull request by JayH5**
- "params" -> "cluster_params" and now all the values can be set
- consul_dns now requires advertise_addr and join parameters
- Update classes to inherit from cluster_params
- Update README

### CODE REVIEW

**Comment by JayH5** [comment]
Next step is to add a class for things that aren't workers or controllers. Things like xylem or router hosts tend to need some stuff like Consul DNS but setting it up is a bit complicated.

We could add a `seed_stack::collaborator` or `member` or something that would use these cluster-wide parameters and set up useful stuff.

**Comment by JayH5** [comment]
Also worth noting that the only thing here that has had a breaking API change is `seed_stack::consul_dns`. The new `seed_stack::cluster_params` is not actually _required_.

**Comment by jerith** [comment]
What's the breaking API change?

**Comment by JayH5** [comment]
`advertise_addr` and `join` are now required parameters (don't default to localhost which was kind of useless)

**Comment by jerith** [comment]
Ah, excellent.

File: manifests/params.pp [hunk 1]
```diff
@@ -1,33 +0,0 @@
...
-
-  $mesos_ensure             = '0.27.1*'
-  $mesos_listen_addr        = '0.0.0.0'
-  $mesos_cluster            = 'seed-stack'
-  $mesos_resources          = {}
-
-  $marathon_ensure          = '0.15.3*'
-  $marathon_default_options = { # TODO
-    'event_subscriber' => 'http_callback' # HTTP callbacks for Consular
-  }
```
Comment (jerith) [inline manifests/params.pp:18]: I assume `$marathon_default_options` wasn't being used anywhere? I can't find any other references to it.

Comment (JayH5) [inline manifests/params.pp:18]: Nope, not used. At one point I wanted all the default things/parameters to be in here but that didn't really ever happen.

**Comment by jerith** [comment]
:+1:

**Comment by jerith** [comment]
:+1:

**Pull request merged**